File: csharp/1328.break-a-palindrome.cs
/*
 * @lc app=leetcode id=1328 lang=csharp
 *
 * [1328] Break a Palindrome
 */

// @lc code=start
public class Solution {
    public string BreakPalindrome(string palindrome) {
        int n = palindrome.Length, h = n / 2, i;
        if (n == 1) {
            return "";
        }
        for (i = 0; i < h && palindrome[i] == 'a'; ++i);
        var ret = palindrome.ToCharArray();
        if (i == h) {
            ret[n - 1] = 'b';
        } else {
            ret[i] = 'a';
        }
        return new string(ret);
    }
}
// @lc code=end

File: java/1328.break-a-palindrome.java
/*
 * @lc app=leetcode id=1328 lang=java
 *
 * [1328] Break a Palindrome
 */

// @lc code=start
class Solution {
    public String breakPalindrome(String palindrome) {
        int n = palindrome.length();
        if (n == 1) {
            return "";
        }
        int i;
        for (i = 0; i * 2 < n - 1&& palindrome.charAt(i) == 'a'; ++i);
        StringBuilder sb = new StringBuilder(palindrome);
        if (i * 2 >= n - 1) {
            sb.replace(n - 1, n, "b");
        } else {
            sb.replace(i, i + 1, "a");
        }
        return sb.toString();
    }
}
// @lc code=end

